Feat/btc gas costs

valeryV99 opened the pull request:
https://www.notion.so/254362757c98814193bdf2683a2f8535?v=254362757c98811fb5a9000c47b44575&p=25d362757c9880cfa97ffb521239a6fd&pm=s&pvs=31

valeryV99 committed:
feat: enhance SwapInput and update dependencies

- Updated `applyMax` logic in `SwapInput` to deduct a fixed fee for BTC and handle edge cases.
- Added unit tests for `SwapInput` to validate `applyMax` behavior, including BTC-specific fee deductions.
- Included new dependencies in `pnpm-lock.yaml`, such as updated versions of `@testing-library/jest-dom`.
- Added `vitest/globals` and `@testing-library/jest-dom` types in `tsconfig.json`.
- Expanded Vitest coverage to include `SwapInput` component files.

valeryV99 committed:
chore: remove unused types in tsconfig.json

- Deleted `vitest/globals` and `@testing-library/jest-dom` types from `tsconfig.json`

opcode0x committed:
chore: updates

valeryV99 committed:
feat: simplify fee calculation and enhance SwapInput tests

- Introduced `calculateAdjustedBalance` utility in `fees.ts` to centralize and simplify BTC fee calculations.
- Updated `SwapInput` to use `calculateAdjustedBalance` for deducing BTC fees, replacing inline logic.
- Enhanced `SwapInput` tests to validate fee calculation using the new utility and dynamic fee rates.
- Added mocks for BTC fee rate in test cases to align with updated logic.

valeryV99 committed:
feat: integrate BTC fee adjustments and enhance balance validation

- Added `calculateAdjustedBalance` utility to centralize BTC fee logic and ensure precise balance calculations.
- Integrated dynamic fee rate using `useBTCFeeRate` in `SwapForm`, `LiquidityForm`, and `SwapInput`.
- Updated balance validation to account for adjusted balances, enhancing accuracy for BTC transactions.
- Added unit tests in `fees.test.ts` to validate `calculateAdjustedBalance` under various scenarios.
- Refactored `applyMax` logic in `SwapInput` to utilize the new `calculateAdjustedBalance` utility.

valeryV99 committed:
test: improve `calculateAdjustedBalance` test cases

- Refactored test cases in `fees.test.ts` for improved readability and coverage.
- Added new scenarios to verify deterministic results, clamping behavior, and monotonicity with respect to fee rates.
- Removed unused `computeFeeWei` helper function to declutter test logic.

opcode0x commented on apps/web/src/shared/lib/fees.ts:
```diff
@@ -0,0 +1,18 @@
+  feeRate: bigint,
+): bigint => {
+  const MIDL_FEE = 3516;
+  const BTC_TX_SIZE = feeRate * 1000n;
```
I think this is incorrect

opcode0x commented on apps/web/src/shared/lib/fees.ts:
```diff
@@ -0,0 +1,18 @@
+  rawBalance: bigint,
+  feeRate: bigint,
+): bigint => {
+  const MIDL_FEE = 3516;
```
MAX MIDL_FEE is 1758, and multiply it by fee rate

valeryV99 committed:
fix: adjust BTC fee calculation logic in `calculateAdjustedBalance`

- Updated `MIDL_FEE` calculation to dynamically scale with the fee rate.
- Adjusted `BTC_TX_FEE` calculation to maintain consistency with dynamic fees.

valeryV99 committed:
test: refactor fee calculation logic in SwapInput tests

- Updated `MIDL_FEE` and `BTC_TX_FEE` constants for clarity and accuracy.
- Adjusted fee calculation to improve test readability and maintain consistency.
